### DISCUSSION TITLE
Add UI configuration option for stripping trailing slashes from result links

### DISCUSSION THREAD

**Discussion by paulrudy** [Pagefind/pagefind#749] [Ideas]
I'd like to suggest a Pagefind UI configuration option to strip trailing slashes in Pagefind results.

I understand that this can be done by passing a function to Pagefind's [processResult](https://pagefind.app/docs/ui/#process-result) config option [as Astro's Starlight does](https://github.com/withastro/starlight/blob/cf12beb91b4cb2f212dbcc0cc1ed56e79d055ff0/packages/starlight/components/Search.astro#L133).

But I think it might be useful enough, to enough people, to have as a standalone configuration option.

**Comment by shishkin** [comment]
Or alternatively, it can be an option for the indexer API.

**Comment by bglw** [comment]
Ha, I've always been a trailing-slash zealot so that bias has certainly crept into Pagefind 😉 

This is a good request, though. I think the ideal would be to replace the [keep-index-url indexing option](https://pagefind.app/docs/config-options/#keep-index-url) with some sort of "URL behavior" option. 

e.g. `--url-format "trailing-slash"` (default) or `--url-format "keep-index"` or `--url-format "strip-trailing-slash"`

**Reply by paulrudy** [reply]
> Ha, I've always been a trailing-slash zealot so that bias has certainly crept into Pagefind 😉 
> 
> 
> 
> This is a good request, though. I think the ideal would be to replace the [keep-index-url indexing option](https://pagefind.app/docs/config-options/#keep-index-url) with some sort of "URL behavior" option. 
> 
> 
> 
> e.g. `--url-format "trailing-slash"` (default) or `--url-format "keep-index"` or `--url-format "strip-trailing-slash"`

Thanks for considering it! I know people can have strong opinions/preferences about trailing slashes but once a decision for a site has been made, it's helpful to have consistency either way.

**Reply by bglw** [reply]
Oh for sure, the worst option is having _both_ available!

**Comment by paulrudy** [comment]
Just checking in whether this idea might be considered soon in a future update?
